src/server/journal/streamer.cc

```diff
@@ -230,14 +230,19 @@ void RestoreStreamer::Run() {
 void RestoreStreamer::SendFinalize(long attempt) {
   VLOG(1) << "RestoreStreamer LSN opcode for : " << db_slice_->shard_id() << " attempt " << attempt;
   journal::Entry entry(journal::Op::LSN, attempt);
+  journal::Entry noop(journal::Op::NOOP, 0);
 
   io::StringSink sink;
   JournalWriter writer{&sink};
   writer.Write(entry);
+
+  // We do flush sending NOOP
+  for (int i = 0; i < 16; ++i) {
+    writer.Write(noop);
+  }
```

Comment:
Is this really the best way to flush a stream? :upside_down_face:

Reply:
we do near the same in replication